Makefile

```diff
@@ -55,9 +59,24 @@ clean_all: clean_volumes clean_orphans clean_images
 # Use this to reset data and ensure only defined services are running
 clean: clean_volumes clean_orphans
 
-# Rebuild and start all services
-re: clean_all up_no_elk
+################################################################################
+# Rebuild and Restart
+################################################################################
+# Rebuild and start all services without removing volumes
+re: clean_orphans clean_images up_no_elk
```

Comment:
The combination of 
```
docker-compose -f ./docker-compose.yml down --remove-orphans
docker rmi -f $$(docker images -aq) 
```
is rather inconsistent, afaik. The 1st command will only stop and remove containers, networks, and orphan containers created by the specific docker-compose.yml file you specify, which is the one in the same directory. However, the 2nd command will attempt to remove all Docker images on your system, regardless of which yaml file or directory they were created from. This can indeed remove images that are being used by other containers not managed by the current yaml file (thus, still up and running). 
So, we should change one of them to be consistent with the other, maybe change the 2nd one so that it will only remove the imgs created by current yaml file, I think.

Reply:
Fixing this was trickier than I thought, but since this target rule was needed, it was important to control what images can be removed. For that, I had to create a temporary file to store image IDs and keep track of it's content until their deletion. Thanks for your eagle eyes at spotting these issues, let's see if you find some more 🐛🔭🦅🙏